autoreject/autoreject.py

```diff
@@ -613,69 +620,128 @@ def transform(self, epochs):
+        _apply_interp(reject_log, self, epochs_clean, self.threshes_,
+                      self.picks_, self.verbose)
+
+        _apply_drop(reject_log, self, epochs_clean, self.threshes_,
+                    self.picks_, self.verbose)
+
+        if return_log:
+            return epochs_clean, reject_log
         else:
-            (_, _, interp_channels, _,
-             bad_epochs_idx, good_epochs_idx) = self._annotate_epochs(
-                 threshes=self.threshes_, epochs=epochs)
-            if len(good_epochs_idx) == 0:
-                raise ValueError('All epochs are bad. Sorry.')
-
-            self._interpolate_bad_epochs(
-                epochs_out, interp_channels=interp_channels,
-                verbose=self.verbose)
-        if np.any(bad_epochs_idx):
-            epochs_out.drop(bad_epochs_idx, reason='AUTOREJECT')
-        else:
-            warnings.warn(
-                "No bad epochs were found for your data. Returning "
-                "a copy of the data you wanted to clean. Interpolation "
-                "may have been done.")
-        return epochs_out
-
-    def _interpolate_bad_epochs(
-            self, epochs, interp_channels, verbose='progressbar'):
-        """Actually do the interpolation."""
-        pos = 4 if hasattr(self, '_leave') else 2
-        for epoch_idx, interp_chs in _pbar(
-                list(enumerate(interp_channels)),
-                desc='Repairing epochs',
-                position=pos, leave=True, verbose=verbose):
-            epoch = epochs[epoch_idx]
-            epoch.info['bads'] = interp_chs
-            interpolate_bads(epoch, picks=self.picks, reset_bads=True)
-            epochs._data[epoch_idx] = epoch._data
+            return epochs_clean
+
+
+def _interpolate_bad_epochs(
+        epochs, interp_channels, picks, verbose='progressbar'):
+    """Actually do the interpolation."""
+    pos = 2  # XXX removed ._leave starte. Perhaps find better heuristic.
+    assert len(epochs) == len(interp_channels)
+
+    for epoch_idx, interp_chs in _pbar(
+            list(enumerate(interp_channels)),
+            desc='Repairing epochs',
+            position=pos, leave=True, verbose=verbose):
+        epoch = epochs[epoch_idx]
+        epoch.info['bads'] = interp_chs
+        interpolate_bads(epoch, picks=picks, reset_bads=True)
+        epochs._data[epoch_idx] = epoch._data
+
+
+def _run_local_reject_cv(epochs, thresh_func, picks_, n_interpolate, cv,
+                         consensus, verbose):
+    n_folds = cv.get_n_splits()
+    loss = np.zeros((len(consensus), len(n_interpolate),
+                    n_folds))
+
+    # The thresholds must be learnt from the entire data
+    local_reject = LocalAutoReject(thresh_func=thresh_func,
+                                   verbose=verbose,
+                                   picks=picks_)
+    local_reject.fit(epochs)
+
+    assert len(local_reject.consensus_) == 1  # works with one ch_type
```

Comment:
So, `LocalAutoReject` works only with one channel type. We should make this class private ...